### ISSUE TITLE
Bug: Timespinner fails to generate on beta-website

### ISSUE THREAD

**Issue by palex00** [ArchipelagoMW/Archipelago#4004]
### What happened?

I tried to generate a website on archipelago.gg:24242 with a YAML made from the same website. Trying to generate gave me an "Internal Server Error"-warning. Generating the same YAML locally works fine.

Any YAML is affected. Scipio tried this:
```
name: TestName
game: Timespinner
Timespinner: {}
```
and it also did not generate.

### What were the expected results?

That it would generate

### Software

Website

**Comment by ScipioWright** [comment]
Additional info:
It fails when generating a single-player game directly from the options page, and also when uploading a yaml to the site.
AFAIK, it's only Timespinner.

**Comment by palex00** [comment]
Local Webhost Error Log:
```
[2024-09-27 19:01:30,560] Exception on /games/Timespinner/generate-yaml [POST]
Traceback (most recent call last):
  File "C:\Users\palex\AppData\Local\Programs\Python\Python311\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\palex\AppData\Local\Programs\Python\Python311\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\palex\AppData\Local\Programs\Python\Python311\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\palex\AppData\Local\Programs\Python\Python311\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Archipelago Stuff\Archipelago-main\WebHostLib\options.py", line 278, in generate_yaml
    return generate_game({player_name: formatted_options})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Archipelago Stuff\Archipelago-main\WebHostLib\options.py", line 55, in generate_game
    return start_generation(options, get_meta({}))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Archipelago Stuff\Archipelago-main\WebHostLib\generate.py", line 86, in start_generation
    options=pickle.dumps({name: vars(options) for name, options in gen_options.items()}),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_pickle.PicklingError: Can't pickle <class 'worlds.timespinner.Options.<class 'worlds.timespinner.Options.StartWithJewelryBox'>Hidden'>: attribute lookup <class 'worlds.timespinner.Options.StartWithJewelryBox'>Hidden on worlds.timespinner.Options failed
```

**Comment by palex00** [comment]
According to Faris this "sounds like a slot data issue then"

**Comment by ScipioWright** [comment]
No, I said it had to do with them putting something into slot data that they shouldn't be

**Comment by ScipioWright** [comment]
From discussion on discord here: https://discord.com/channels/731205301247803413/731214280439103580/1289301564556709940
It is the f-string in OptionHider that is causing this

**Comment by Berserker66** [comment]
It is neither the f-string not slot data.
HoweverI posted a potential solution here:
https://discord.com/channels/731205301247803413/731214280439103580/1289329864641347637
That works for everything except DeathLink

**Comment by Jarno458** [comment]
@here plz have a look at #4050 as that should hopefully be able to close this issue

**Comment by Jarno458** [comment]
Fixed in #4050
